s3tests/functional/test_s3_website.py

```diff
@@ -1081,4 +1081,4 @@ def test_routing_generator():
     for t in ROUTING_RULES_TESTS:
         if 'xml' in t and 'RoutingRules' in t['xml'] and len(t['xml']['RoutingRules']) > 0:
             t['xml']['RoutingRules'] = common.trim_xml(t['xml']['RoutingRules'])
-        yield routing_check, t
+    routing_check(t)
```

Comment:
This is the deprecated feature behind this:
https://docs.pytest.org/en/stable/deprecations.html#yield-tests

What I do, is just call the test again and again, which is not as elegant as a parameterization would be, but it also means less code, and I assume this will eventually be fixed upstream, as this should eventually stop working.

My commit did have an issue still however, the indentation was wrong, which I've fixed now.